server/rpc/interceptors/interceptors.go

```diff
@@ -195,6 +195,28 @@ func roleAuthUnaryServerInterceptor(env environment.Env) grpc.UnaryServerInterce
 	}
 }
 
+func ipAuthUnaryServerInterceptor(env environment.Env) grpc.UnaryServerInterceptor {
+	return func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
+		if irs := env.GetIPRulesService(); irs != nil {
+			if err := irs.Authorize(ctx); err != nil {
+				return nil, err
+			}
+		}
+		return handler(ctx, req)
+	}
+}
+
+func ipAuthStreamServerInterceptor(env environment.Env) grpc.StreamServerInterceptor {
+	return func(srv interface{}, stream grpc.ServerStream, info *grpc.StreamServerInfo, handler grpc.StreamHandler) error {
+		if irs := env.GetIPRulesService(); irs != nil {
+			if err := irs.Authorize(stream.Context()); err != nil {
+				return err
+			}
+		}
+		return handler(srv, stream)
+	}
+}
+
```

Comment:
have you tried running `cacheload` against a local server w/ IP auth enabled to see whether it significantly affects throughput?

Reply:
I have not, but is there a reason to believe it would affect perf? Apart from loading IP rules from DB on a cache miss, the rest of the operations are fast.

Reply:
I agree it would be nice to do a quick check here just in case. It's relatively cheap to do and will let us know we didn't miss something silly (like if the cache were busted somehow and we were reading from DB every time or something)

Reply:
95% latency, in milliseconds:
![rule_check_latency](https://github.com/buildbuddy-io/buildbuddy/assets/1335358/9a7a4929-2d6d-4e96-b54f-f09fa7bc8371)

Reply:
and now it's down to 0.01 ms after fixing the unnecessary group lookup